### PR TITLE
[Change] More complex unit tests

### PR DESCRIPTION
**Pull request by matthieun**
### Description:

- Rename Bloated features `shallowFromXXXX` to `shallowFrom`, and `fromXXXX` to `from`
- Added convenience methods in BloatedNode to replace in/out edge identifiers
- Formalized `AtlasChangeGenerator` as an interface for anything that generates feature change sets from an Atlas. It uses ChangeAtlas directly, and its suite of validators to validate the change sets.

### Potential Impact:

N/A

### Unit Test Approach:

Added complex unit tests:
- Split some edges in half
- Alter node tags in some cases
- Remove all reverse edges

All new tests are generated on a smaller Atlas, but then applied on a bigger one to check for boundary effects. 

### Test Results:

Tests pass

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChange.java [hunk 1]
```diff
@@ -51,6 +62,18 @@ public Rectangle bounds()
         return this.reference.bounds();
     }
 
+    @Override
+    public boolean equals(final Object other)
```
Comment (lucaspcram) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChange.java]: Great! I was going to add this as part of my PR. Glad you already did. We will need one for the `Change` class as well.

File: src/main/java/org/openstreetmap/atlas/geography/atlas/change/AtlasChangeGenerator.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.utilities.conversion.Converter;
+
+/**
+ * Something that takes an {@link Atlas} and produces a set of {@link FeatureChange} that should be
+ * apply-able back to the initial {@link Atlas}
+ *
+ * @author matthieun
+ */
+public interface AtlasChangeGenerator extends Converter<Atlas, Set<FeatureChange>>
```
Comment (lucaspcram) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/change/AtlasChangeGenerator.java]: This is a great idea. Making reusable templates is going to be a huge win.

File: src/test/java/org/openstreetmap/atlas/geography/atlas/change/AtlasChangeGeneratorSplitRoundabout.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.openstreetmap.atlas.geography.atlas.bloated.BloatedNode;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.utilities.collections.Maps;
+import org.openstreetmap.atlas.utilities.collections.Sets;
+
+/**
+ * @author matthieun
+ */
+public class AtlasChangeGeneratorSplitRoundabout implements AtlasChangeGenerator
```
Comment (lucaspcram) [inline src/test/java/org/openstreetmap/atlas/geography/atlas/change/AtlasChangeGeneratorSplitRoundabout.java]: 👍

File: src/main/java/org/openstreetmap/atlas/geography/atlas/bloated/BloatedLine.java [hunk 4]
```diff
@@ -153,6 +153,22 @@ public BloatedLine withRelationIdentifiers(final Set<Long> relationIdentifiers)
         return this;
     }
 
+    public BloatedLine withTagExtra(final String key, final String value)
```
[the remark this answers is not in the document]
Comment (matthieun) [reply src/main/java/org/openstreetmap/atlas/geography/atlas/bloated/BloatedLine.java]: Options are always open!

**Pull request merged**